Computer History Museum Archive.js

```diff
@@ -0,0 +1,1000 @@
+{
+	"translatorID": "f3b81c4e-28b4-41ae-9824-55739fe9c91a",
+	"label": "Computer History Museum Archive",
+	"creator": "Bo An",
+	"target": "^https?://www.computerhistory.org/collections/",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2020-12-03 17:02:51"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+	Copyright © 2020 Bo An
+	This file is part of Zotero.
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+	***** END LICENSE BLOCK *****
+*/
+
+
+
+function detectWeb(doc, url) {
+	const collection = "/collections/search/";
+	const artifact = "/collections/catalog/"
+	const isCollection = url.includes(collection)
+	const isArtifact = url.includes(artifact)
+	if (isCollection) {
+		return 'multiple';
+	}
+	else if (isArtifact) {
+	//todo it'd be better to use different items types according to item types in the collection, i.e. interview, video, audio etc, but for now 'book' has most fields needed.
+		return 'book';
+	}
+	return false;
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) === "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), (items) => {
+			if (!items) {
+				return true;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+			return true;
+		});
+	}
+	else {
+		scrape(doc, url);
+	}
+}
+
+function getSearchResults(doc, checkOnly) {
+	let items = {};
+	let found = false;
+	const entries = doc.querySelectorAll('p.objtext');
+
+	for (let i = 0; i < entries.length; i++) {
+		const titleDiv = entries[i].querySelector('span.objtitle a');
+		if (!titleDiv) continue;
+		const href = titleDiv.href;
+		const title = titleDiv.textContent;
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+function scrape(doc, url) {
+	const fields = [...doc.querySelectorAll("div.field")];
+
+	let newItem = new Zotero.Item("book");
+	newItem.archive = 'Computer History Museum';
+
+	fields.forEach((nodeOriginal) => {
+		// clone the node to avoid altering the document.
+		const node = nodeOriginal.cloneNode(true)
```

Comment:
Please lint and particularly add semicolons!

Reply:
Hi Abe, Thanks for the comments. I have fixed others but this. For some reason, the repo ESLINT config is not working on IDEs (Intellij, VS Code) as they show ESLINT no header error. Do you know how to fix this? If not, don't bother, I'll try lint by other means. Thx.

Reply:
It's working for me in VS Code, but you need to have installed the repo dependencies with `npm i`, and you need to give VS Code permission to load the local installation when you first click on ESLINT in the bottom right:

![eslint-validation](https://user-images.githubusercontent.com/783082/120152303-4ffda480-c1bb-11eb-98e2-81c904821db4.png)

If you previously installed dependencies but don't remember what you chose for that, you can run "ESLint: Reset Library Execution Decisions" to force it to ask you again.

Reply:
Thanks @dstillman! I just got it working but in a very hacky way. It turned out to be an weird bug. I still don't know the exact cause.

When I run `eslint "computer history museum archive.js"`, the "no header" error appears because of these two lines:

https://github.com/zotero/translators/blob/809e3973697036a2d0ccb3fbfa7b96c585c4168e/.ci/eslint-plugin-zotero-translator/lib/translators.js#L109

```
	let m = source.match(re.undecorated);
	if (!m) throw new Error('no header');
```

And then the script is run NOT on the "computer history museum archive.js", but on the very first translator file, i.e. A Contra Corriente.js (if you remove it, the next will be `Ab Impreio.js.` with the same error.), in the folder by alphabetic order. In other words, the "no header" error is raised when checking all the translator files and it failed on the very first one.

However, t**he strange thing is** that when I checked both `source` (the text of the linted translator file) and the regular expression `re.undecorated`, **they both have the correct values.** And if I match them in an empty node script  by calling `source.match(re)` or in a regular expression tester, it returns the correct matched array. 

- Regex Tester
![image](https://user-images.githubusercontent.com/7269717/120224078-0062bb80-c243-11eb-85dc-d272e24217dc.png)


But **when ESLink calls it, it returns "Null" as the result**, which raises the exception in the second line.


- Test Code:
![image](https://user-images.githubusercontent.com/7269717/120224117-140e2200-c243-11eb-976d-166264a5a336.png)


- Console output:
![image](https://user-images.githubusercontent.com/7269717/120223937-b5e13f00-c242-11eb-90a4-f0bcad359060.png)


So the only way for me to run lint on my file is to bypass this function by `return decorated` in all cases, and  two other steps. 1. rename the file into one without empty spaces (otherwise it will raise a separate `TypeError: Cannot read property 'source' of undefined` error). 2. Comment out the beginning meta translator information section. 

 
![image](https://user-images.githubusercontent.com/7269717/120224841-803d5580-c244-11eb-9c83-e10351b68455.png)

# In short

**For unknown reasons, `let m = source.match(re.undecorated);` returned "null" for m even though both `source` and `re.undecorated` have the correct text and pattern.** And it caused the parser for the translator files and eventually ESLint to fail.

Reply:
A possible explanation: https://stackoverflow.com/questions/54607777/string-match-function-returns-null-even-when-there-is-a-match

Reply:
OK, linting looks good now. (The comment on line 42 isn't quite indented enough, but that doesn't really matter and doesn't cause a CI failure.)